src/ast.js

```diff
@@ -130,12 +136,32 @@ class Literal {
     return this.length === 1;
   }
 
+  get isUnicode() {
+    return this.flags && this.flags.indexOf('u') !== -1;
+  }
+
   get length() {
     return this.value.length;
   }
 
   toString() {
-    return jsesc(this.value).replace(/([\t\n\f\r\$\(\)\*\+\-\.\?\[\]\^\{\|\}])/g, '\\$1');
+    return jsesc(this.value, { es6: this.isUnicode })
+      .replace(/[\t\n\f\r\$\(\)\*\+\-\.\?\[\]\^\|]/g, '\\$&')
+      .replace(/[\{\}]/g, (match, offset, string) => {
+        // special handling to not escape curly braces which are part of Unicode escapes
+
+        // don't escape opening curly braces immediately preceded by `\u`
+        if (match === '{' && string.slice(offset - 2, offset) === '\\u') {
+          return match;
+        }
+
+        // don't escape closing curly braces ending Unicode escapes
+        if (match === '}' && string.slice(offset - 8, offset - 5) === '\\u{') {
+          return match;
```

Comment:
I think this won't always work since unicode escapes can be variable length, e.g. `\u{10ffff}`. Maybe try something like this:

```javascript
.replace(/(\\u\{[a-z1-9]+\})|([\{\}])/ig, (m, u, e) => u || '\\' + e)
```